spec/02-integration/08-status_api/03-readiness_endpoint_spec.lua

```diff
@@ -0,0 +1,184 @@
+local helpers = require "spec.helpers"
+
+for _, strategy in helpers.all_strategies() do
+  local describe_func = pending
+  if strategy ~= "off" then
+    -- skip the "off" strategy, as dbless has its own test suite
+    describe_func = describe
+  end
+
+  describe_func("Status API - with strategy #" .. strategy, function()
+    local status_client
+    local admin_client
+    lazy_setup(function()
+      helpers.get_db_utils(nil, {})
+      assert(helpers.start_kong {
+        status_listen = "127.0.0.1:8100",
+        plugins = "admin-api-method",
+        database = strategy,
+        nginx_worker_processes = 8,
+      })
+      status_client = helpers.http_client("127.0.0.1", 8100, 20000)
+
+      admin_client = helpers.admin_client()
+    end)
+
+    lazy_teardown(function()
+      if status_client then
+        status_client:close()
+      end
+      assert(helpers.stop_kong())
+    end)
+
+    describe("status readiness endpoint", function()
+
+      it("should return 200 in db mode", function()
+        local res = assert(status_client:send {
+          method = "GET",
+          path = "/status/ready",
+        })
+        assert.res_status(200, res)
+
+      end)
+
+      it("should return 200 after loading an invalid config following a previously uploaded valid config.", function()
+        local res = assert(status_client:send {
+          method = "GET",
+          path = "/status/ready",
+        })
+
+        assert.res_status(200, res)
+
+        local res = assert(admin_client:send {
+          method = "POST",
+          path = "/config",
+          body = {
+            config = [[
+              _format"\!@#$
+            ]]
+          },
+          headers = {
+            ["Content-Type"] = "multipart/form-data"
+          },
+        })
+
+        assert.res_status(400, res)
+
+        ngx.sleep(3)
```

Comment:
Is this `sleep` necessary? is it possible for using [this](https://github.com/Kong/kong/blob/40008ec9a1ad423acc97912e0290737e1daea867/spec/02-integration/01-helpers/01-helpers_spec.lua#L105-L112) instead.